File: 0283-move-zeroes/0283-move-zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        std::ios_base::sync_with_stdio(false);
        std::cin.tie(nullptr);
        int k=0;
        for(int i=0;i<nums.size()-1;i++){
            for(int j=i+1;j<nums.size();j++){
                if(nums[i]==0&&nums[j]!=0){
                    k=nums[i];
                    nums[i]=nums[j];
                    nums[j]=k;
                    break;
                }
            }
        }
        
    }
};

File: 0283-move-zeroes/0283-move-zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        n = len(nums)
        for i in range(n - 1):
            j = i + 1
            if nums[i] == 0:
                while j < n and nums[j] == 0:
                    j += 1
                if j < n:  # Ensure j is within bounds
                    nums[i], nums[j] = nums[j], nums[i]
